src/likelihoods/categorical.jl

```diff
@@ -1,22 +1,52 @@
 """
-    CategoricalLikelihood(l=softmax)
+    CategoricalLikelihood(l=softmax; bijective::Union{Bool,Val{false},Val{true}}=Val(true))
 
 Categorical likelihood is to be used if we assume that the 
-uncertainity associated with the data follows a Categorical distribution.
+uncertainty associated with the data follows a [Categorical distribution](https://en.wikipedia.org/wiki/Categorical_distribution) with `n` categories.
 ```math
     p(y|f_1, f_2, \\dots, f_{n-1}) = \\operatorname{Categorical}(y | l(f_1, f_2, \\dots, f_{n-1}, 0))
+    p(y|f_1, f_2, \\dots, f_n) = \\operatorname{Categorical}(y | l(f_1, f_2, \\dots, f_n))
 ```
+Two variants are possible (you can use a `Bool` or `Val{true}`/`Val{false}` but
+we recommend the latter for type stability).
+## `bijective=true/Val{true}`
 Given an `AbstractVector` ``[f_1, f_2, ..., f_{n-1}]``, returns a `Categorical` distribution,
 with probabilities given by ``l(f_1, f_2, ..., f_{n-1}, 0)``.
+It is used by default.
+
+## `bijective=false`
+Given an `AbstractVector` ``[f_1, f_2, ..., f_{n}]``, returns a `Categorical` distribution,
+with probabilities given by ``l(f_1, f_2, ..., f_{n})``.
+This variant is over-parametrized, as there are `n-1` independent parameters 
+embedded in a `n` dimensional parameter space.
+For more details, see the end of the section of this [Wikipedia link](https://en.wikipedia.org/wiki/Exponential_family#Table_of_distributions)
+where it corresponds to Variant 1 and 2.
 """
-struct CategoricalLikelihood{Tl<:AbstractLink} <: AbstractLikelihood
+struct CategoricalLikelihood{Tb,Tl<:AbstractLink} <: AbstractLikelihood
     invlink::Tl
+    CategoricalLikelihood{Tb}(invlink) where {Tb} = CategoricalLikelihood{Tb}(Link(invlink))
+    CategoricalLikelihood{Tb}(invlink::Tl) where {Tb,Tl<:AbstractLink} = new{Tb,Tl}(invlink)
```

Comment:
Alternatively, one could use
```suggestion
    function CategoricalLikelihood{Tb}(invlink) where {Tb}
        l =  Link(invlink)
        new{Tb,typeof(l)}(l)
    end
```

Reply:
But with your proposition if one passes a `Link` like `SoftMaxLink` it's going to be wrapped twice.

Reply:
This issue is dealt with in #65